TEveEventDisplay/src/TEveMu2eMainWindow.cc

```diff
@@ -862,6 +863,7 @@ namespace mu2e{
       _firstLoop = firstLoop;
       _accumulate = accumulate;
       _data.chcol = data.chcol;
+      _data.tccol = data.tccol;
```

Comment:
Please make a new PR that addresses the comments below.  

Are you copying the entire content of data to _data?  If so, the right way is to use the assignment operator of the class Data_Collections .  The one that is automatically generated by the compiler will do the right thing.  Again, I now that this is code that you inherited.  I see that the assignment operator is declared in this code but it I don't see where it is defined.  Does it actually have a definition?  If not, I *think* that the declaration blocks the compiler from providing the default one.

And a second comment. In the signature of TEveMu2eMainWindow::setEvent, the argument data is passed by reference, which is good.  But it is not const reference.  Is it actually modified by this code?  If not please make it const reference.  If you can't, because lower level code is not const correct, please fix that too.

Reply:
The variable _data is defined in the header file "TEveMu2eMainWindow.h". 

 I understand your comment but I am sorry since I followed the pattern that was set previously for functions like AddComboHits() or AddCrystalHits() while developing these new functions, I think @sophiemiddleton would be able to answer/explain this much better.

Reply:
I will check. There is a work around in that code to allow for add/remove tick box on the GUI and that data_ variable is part of that. It's a bit "hacky" but getting TEve to do what you want is a bit that way in general

Reply:
> The variable _data is defined in the header file "TEveMu2eMainWindow.h".
> 
> I understand your comment but I am sorry since I followed the pattern that was set previously for functions like AddComboHits() or AddCrystalHits() while developing these new functions, I think @sophiemiddleton would be able to answer/explain this much better.

I understand that you followed an existing pattern.  That's OK, even expected.  I would like to evolve towards better patterns as we move ahead, particularly was we move into REve.

It does look to me as if you are copying the entire content of _data.  Am I right?   If so you should do that in one line by implementing the correct member function in the class.  Do you know how to do that?

Also fix const correctness when you can - doing it properly needs to be done from the bottom up so it's tedious.

Reply:
> I will check. There is a work around in that code to allow for add/remove tick box on the GUI and that data_ variable is part of that. It's a bit "hacky" but getting TEve to do what you want is a bit that way in general.

I am talking about the data members defined here:
https://github.com/NamithaChitrazee/Offline-1/blob/42715230c5fff46f969b7658f457b79cdac601a8/TEveEventDisplay/src/dict_classes/Data_Collections.h#L46

All are bare pointers to data products. It's possible to read a file that does not contain one of these data products, so the pointers can stay zero.  You should use a type of pointer that will throw an exception if you forget to check for validity and attempt to dereference an invalid pointer.

Reply:
Yes, I know we need to rewrite this. I have rewritten it a bit for REve but further advice is welcome (please email my with suggestions so I dont lose track)